doc/tutorials/understanding-error-messages.md

```diff
@@ -0,0 +1,48 @@
+# Understanding Error Messages
+
+If you want to use Elektra for any of your projects you will sooner or later run into errors. These can happen
+quite easily such as trying to mount with insufficient permissions, missing plugin configurations or when you want to write
+specifications for your configurations and test it out.
+
+Here is an example how you can trigger an error for an invalid IP entry in the `hosts` file:
+
+```sh
+sudo kdb mount --with-recommends /etc/hosts system/hosts hosts
+sudo kdb set system/hosts/ipv4/my 88.198.134.1777
+# RET: 5
+# Sorry, module network issued the error C03200:
+# Validation Semantic: name: system/hosts/ipv4/my value: 88.198.134.1777 message: Name or service not known
+
+#cleanup
+sudo kdb umount system/hosts
+```
```

Comment:
Now continue explaining all parts that are involved to make this error like it is and link to all deepening documents about errors (in dev and decisions).

Reply:
I am still unsure what you else want. The `categorization guideline` actually contains all necessary information about the errors. I can also link to the error message format but what else do you think of?

Reply:
How to emit errors (e.g. in plugins), how to handle errors in tools, how to generate error message (-d, -v), ...